### DISCUSSION TITLE
Gitlab auto approve

### DISCUSSION THREAD

**Discussion by artemptushkin** [renovatebot/renovate#23277] [Suggest an Idea]
### Tell us more.

The property [autoapprove](https://docs.renovatebot.com/configuration-options/#autoapprove) is available only for the Azure platform.

Though it's a valuable feature for others, for example, if you have more than one approval required you can set one auto-approve from the Renovate bot.

It Aitlab it can be done with calling the API https://docs.gitlab.com/ee/api/merge_request_approvals.html#approve-merge-request

**Comment by rarkins** [comment]
Assuming that Renovate is running as a regular user with PAT, is it allowed to approve its own merge requests?

**Reply by viceice** [reply]
github and gitea forbid approving own PR's

**Reply by artemptushkin** [reply]
yes, it's a setting https://docs.gitlab.com/ee/user/project/merge_requests/approvals/settings.html#prevent-approval-by-author

I just implemented this feature but with a custom job using Renovate token

**Reply by artemptushkin** [reply]
@rarkins any chance this will be taken into the Renovate backlog?

**Reply by rarkins** [reply]
The docs indicate that gitlab is supported: 
![image](https://github.com/renovatebot/renovate/assets/6311784/0f8991ec-7163-4cab-88f9-5f558beb7089)

Please take a look and clarify what difference you are asking for

**Reply by artemptushkin** [reply]
@rarkins appreciate it, the discussion is closed then!
